controlplane/src/db/schema.ts

```diff
@@ -595,9 +595,93 @@ export const organizations = pgTable('organizations', {
     .references(() => users.id)
     .notNull(),
   createdAt: timestamp('created_at', { withTimezone: true }).notNull().defaultNow(),
-  isPersonal: boolean('is_personal').default(false),
-  isFreeTrial: boolean('is_free_trial').default(false),
-  isRBACEnabled: boolean('is_rbac_enabled').notNull().default(false),
+  /**
+   * @deprecated
+   */
+  isPersonal: boolean('is_personal'),
+  /**
+   * @deprecated
+   */
+  isFreeTrial: boolean('is_free_trial'),
+  /**
+   * @deprecated
+   */
+  isRBACEnabled: boolean('is_rbac_enabled'),
+});
+
+export const organizationBilling = pgTable(
+  'organization_billing',
+  {
+    id: uuid('id').notNull().primaryKey().defaultRandom(),
+    organizationId: uuid('organization_id')
+      .notNull()
+      .references(() => organizations.id, {
+        onDelete: 'cascade',
+      }),
+    plan: text('plan'),
+    email: text('email'),
+    stripeCustomerId: text('stripe_customer_id'),
+  },
+  (t) => {
+    return {
+      orgIndex: uniqueIndex('organization_billing_idx').on(t.organizationId),
+      stripeIndex: uniqueIndex('organization_billing_stripe_idx').on(t.stripeCustomerId),
+    };
+  },
+);
+
+export type Feature = {
+  id: string;
+  description?: string;
+  limit?: number;
+};
+
+export const billingPlans = pgTable('billing_plans', {
+  id: text('id').notNull().primaryKey(),
+  active: boolean('active').notNull().default(true),
+  name: text('name').notNull(),
+  price: integer('price').notNull(),
+  features: customJson<Feature[]>('features').notNull(),
+  stripePriceId: text('stripe_price_id'),
+  weight: integer('weight').notNull().default(0),
+});
+
+// These statuses map directly to Stripe's subscription statuses
+const statuses = [
+  'incomplete',
+  'incomplete_expired',
+  'trialing',
+  'active',
+  'past_due',
+  'canceled',
+  'unpaid',
+  'paused',
+] as const;
+
+export const subscriptionStatusEnum = pgEnum('status', statuses);
+
+export type SubscriptionStatus = (typeof statuses)[number];
+
+export const billingSubscriptions = pgTable('billing_subscriptions', {
+  id: text('id').notNull().primaryKey(),
+  organizationId: uuid('organization_id')
+    .notNull()
+    .references(() => organizations.id, {
+      onDelete: 'cascade',
+    }),
+  metadata: customJson<{ [key: string]: string }>('metadata').notNull(),
+  status: subscriptionStatusEnum('status').notNull(),
+  priceId: text('price_id').notNull(),
+  quantity: integer('quantity').notNull(),
+  cancelAtPeriodEnd: boolean('cancel_at_period_end').notNull(),
+  cancelAt: timestamp('cancel_at', { withTimezone: true }),
```

Comment:
Three different cancel? :smile: Could you add a comment?

Reply:
I'll link to the stripe docs, as above. These fields map 1 on 1.

Reply:
Sounds good